litestar/events/emitter.py

```diff
@@ -84,12 +86,16 @@ async def _worker(receive_stream: MemoryObjectReceiveStream) -> None:
         Returns:
             None
         """
-        async with receive_stream, anyio.create_task_group() as task_group:
+        async with receive_stream:
             async for item in receive_stream:
-                fn, args, kwargs = item
-                if kwargs:
-                    fn = partial(fn, **kwargs)
-                task_group.start_soon(fn, *args)
+                try:
+                    async with anyio.create_task_group() as task_group:
+                        fn, args, kwargs = item
+                        if kwargs:
+                            fn = partial(fn, **kwargs)
+                        task_group.start_soon(fn, *args)
+                except Exception as exc:
+                    logger.exception("Error in event listener: %s", exc)
```

Comment:
In addition to log letting exceptions propagate into the receive stream iterator, it makes sense to have a task group per item as running all events in the same task group would result in the cancellation of any unfinished listeners when any one of them raises.